File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """

        rev, orig = 0, x

        while x > 0:
            n = orig % 10
            rev = (rev * 10) + (x % 10)
            x //= 10
        

        return rev == orig
        

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int rev = 0;
        int cp = x;
        while(cp > 0){

            // 121 % 10 = 1
            // 121 / 10 =  12
            // (0 * 10) + 1 = 1
            // 12 % 10 = 2
            // 12 / 10 = 1
            // (1 * 10) + 2 = 12
            // 1 % 10 = 1
            // 1 / 10 = 0
            // (12 * 10) + 1 = 121 
            
            int rem = cp % 10;
            rev = (rev * 10) + rem;
            //update
            cp /= 10; 
        }
        return x == rev;
    }
}